### PR TITLE
SD card salt

### PR DESCRIPTION
**Pull request by andrewkozlik**
Implements https://github.com/trezor/trezor-firmware/issues/307.

### CODE REVIEW

File: common/protob/messages-management.proto [hunk 1]
```diff
@@ -131,6 +131,24 @@ message ChangePin {
...
+ * @next Success
+ * @next Failure
+ */
+message SdSalt {
+    optional SdSaltOperationType operation = 1;
+    /**
+    * Structure representing SD card salt operation
+    */
+    enum SdSaltOperationType {
+        ENABLE = 0;
```
Comment (prusnak) [inline common/protob/messages-management.proto]: This is just a nitpick. but I'd say `DISABLE = 0` and `ENABLE = 1` make more sense.

File: storage/storage.h [hunk 2]
```diff
@@ -24,6 +24,9 @@
 #include <stdint.h>
 #include "secbool.h"
 
+// The length of the external salt in bytes.
+#define EXTERNAL_SALT_SIZE 32
```
Comment (prusnak) [inline storage/storage.h]: Minimum block size on the SD card is 512 bytes. Surely it makes no sense to use 4096 bits of entropy, but we might consider increasing this to 64 bytes?

File: python/src/trezorlib/cli/trezorctl.py [hunk 3]
```diff
@@ -267,6 +275,22 @@ def change_pin(connect, remove):
...
 
+@cli.command()
+@click.argument("operation", type=CHOICE_SD_SALT_OPERATION_TYPE)
+@click.pass_obj
+def sd_salt(connect, operation):
+    """Configure SD card salt protection.
+
+    Enable, disable or regenerate SD card salt. The options are:
+
+    \b
```
Comment (prusnak) [inline python/src/trezorlib/cli/trezorctl.py]: What does `\b` mean in this context?

**Comment by prusnak** [comment]
CI build failure for legacy fixed in https://github.com/trezor/trezor-firmware/tree/andrewkozlik/sd-salt-squashed-fixed branch. There are still outstanding `test core` failures in the CI.

File: python/src/trezorlib/cli/trezorctl.py (hunk 3 above)
Comment (andrewkozlik) [reply python/src/trezorlib/cli/trezorctl.py]: By default Click rewraps text based on the current width of the terminal. \b disables this behavior, i.e. it forces Click to preserve newlines.

File: common/protob/messages-management.proto (hunk 1 above)
Comment (andrewkozlik) [reply common/protob/messages-management.proto]: ACK

File: python/src/trezorlib/cli/trezorctl.py (hunk 3 above)
Comment (prusnak) [reply python/src/trezorlib/cli/trezorctl.py]: TIL

File: storage/storage.h (hunk 2 above)
Comment (andrewkozlik) [reply storage/storage.h]: Well, I think that when the salt enters PBKDF2-HMAC-SHA256, the entropy eventually gets hashed down to 256 bits anyway. So using a higher value here would be giving a false sense of greater security.

Comment (prusnak) [reply storage/storage.h]: OK

**Comment by andrewkozlik** [comment]
Replaced by https://github.com/trezor/trezor-firmware/pull/526.

**Pull request closed without merging**